### ISSUE TITLE
vmagent will grab a lot of deleted pods

### ISSUE THREAD

**Issue by guidao** [VictoriaMetrics/VictoriaMetrics#1830]
**Describe the bug**
vmagent will grab a lot of deleted pods

**Expected behavior**
No similar errors will occur

**Screenshots**
![企业微信截图_484e784f-b5ca-4827-a0cf-ac6526b9da28](https://user-images.githubusercontent.com/3192136/142361979-bf698c4e-669c-4155-9ac9-854ac6f2929c.png)

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
/ # /vmagent-prod --version
vmagent-20211108-135505-tags-v1.69.0-0-g69e655ba7
```

**Comment by f41gh7** [comment]
Possible workaround reduce value for flag: - `-promscrape.kubernetesSDCheckInterval` to 5s for example. Default value is 30s.

 vmagent  refreshes list of targets according to this flag.

**Comment by guidao** [comment]
The same pod did not disappear even after waiting for more than 30s.

**Comment by guidao** [comment]
Another problem, we found that if the ip of the pod is reused, it will grab it twice. It will be deleted after restarting vmagent.
![image](https://user-images.githubusercontent.com/3192136/142578909-92948629-0903-444e-a7dd-428cab6aabdd.png)

**Comment by valyala** [comment]
FYI, the bugfix has been included in [v1.70.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases)
